MonstieWash/Assets/Scripts/Audio/MonsterSoundManager.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(SoundPlayer))]
+public class MonsterSoundManager : MonoBehaviour
+{
+    [System.Serializable]
+    private struct MoodSound
+    {
+        public MoodType mood;
+        public Sound sound;
+    }
+
+    [SerializeField] private List<MoodSound> moodSounds = new();
+    [SerializeField] private List<Sound> attackSounds = new();
+
+    private SoundPlayer m_soundPlayer;
+    private MonsterBrain m_monsterBrain;
+    private MusicManager m_musicManager;
+
+    private void Awake()
+    {
+        m_soundPlayer = GetComponent<SoundPlayer>();
+        m_monsterBrain = GetComponent<MonsterBrain>();
+        m_musicManager = FindFirstObjectByType<MusicManager>();
+    }
+
+    private void OnEnable()
+    {
+        m_monsterBrain.OnMoodChanged += PlayMoodSound;
+        m_monsterBrain.OnMoodChanged += UpdateMusic;
+    }
+
+    private void OnDisable()
+    {
+        m_monsterBrain.OnMoodChanged -= PlayMoodSound;
+        m_monsterBrain.OnMoodChanged -= UpdateMusic;
+    }
+
+    /// <summary>
+    /// Plays a random attack sound. Will be updated when attacks are implemented.
+    /// </summary>
+    public void PlayAttackSound()
+    {
+        var index = Random.Range(0, attackSounds.Count - 1);
+
+
+        m_soundPlayer.SwitchSound(attackSounds[index]);
+        m_soundPlayer.PlaySound(true);
+    }
+
+    /// <summary>
+    /// Plays the sound corresponding to the new current mood.
+    /// </summary>
+    /// <param name="mood">The mood that is now active.</param>
+    private void PlayMoodSound(MoodType mood)
+    {
+        var sound = moodSounds.Find(ms => ms.mood == mood).sound;
+        if (sound == null) return;
+
+        m_soundPlayer.SwitchSound(sound);
+        m_soundPlayer.PlaySound(true);
+    }
+
+    /// <summary>
+    /// Escalates or de-escalates the music based on the new mood.
+    /// </summary>
+    /// <param name="mood">The mood that is now active.</param>
+    private void UpdateMusic(MoodType mood)
+    {
+        switch (mood.MoodName)
+        {
+            case "Happy": m_musicManager.SetMusic(MusicManager.MusicType.Background); break;
```

Comment:
If hard-coding is a temporary solution feel free to dismiss this, but music behavior based on moods changing seems like something the designers would need to tinker with so maybe hard-coding isn't the best idea here. 
Regardless, I think the Idle mood should also reset the music to the lowest version, in addition to the Happy mood. Currently, the monster returning to idle would escalate the music which doesn't make much sense (to me at least)

Reply:
Yeah this is all temporary just to test that it would work, the actual dynamic music is for next sem. I'll make a note of that in the script.